module4/ex24_uniquish/ex24_uniquish.py

```diff
@@ -2,9 +2,23 @@ class Foo:
     def __init__(self, x):
         self.x = x
 
+    def __eq__(self, other):
+        if type(other) is type(self):
+            return self.x == other.x
+        return False
+
+    def __hash__(self):
+        return hash(self.x)
+
 
 class Uniquish:
-    pass
+    def __eq__(self, other):
+        if type(self) == type(other):
+            return self.__dict__ == other.__dict__
+
+    def __hash__(self):
+        my_str_list = [str(item) for item in self.__dict__.values()]
+        return hash(tuple(my_str_list))
```

Comment:
Good for converting to string so to make sure the item is hashable.
Nothing to change here, I would've include also attribute name, value type along with the value to make sure about  a more in-depth equality.
Something like this would work:
hash(tuple((attr, str(value), type(value)) for attr, value in self.__dict__ .items()))